sqlmesh/core/loader.py

```diff
@@ -538,9 +538,14 @@ def _model_cache_entry_id(self, model_path: Path) -> str:
                 [
                     str(max(m for m in mtimes if m is not None)),
                     self._loader._context.config.fingerprint,
-                    # We need to check default catalog since the provided config could not change but the
-                    # gateway we are using could change, therefore potentially changing the default catalog
-                    # which would then invalidate the cached model definition.
+                    # default catalog can change outside sqlmesh (e.g., DB user's
+                    # default catalog), and it is retained in cached model's fully
+                    # qualified name
                     self._loader._context.default_catalog or "",
+                    # gateway can change at runtime, and it is retained in a cached
```

Comment:
what do you mean gateway can change at runtime?

Reply:
`sqlmesh --gateway diff_gateway plan`

Reply:
I guess "runtime" isn't very clear in this context - maybe "execution time (e.g., by passing gateway to a CLI command)"?

Reply:
it's more just gateway is configurable